docs/source/aggregation.rst

```diff
@@ -0,0 +1,187 @@
+
+.. code-block:: python
+
+    @aggregator.groupsof(2)
+    @Project.operation
+    def op4(job1, job2):
+        pass
+
+In the above example, the jobs will get aggregated in groups of 2 and hence, only two jobs will
+be passed as parameters at once.
+
+.. note::
+
+    In case the number of jobs in the project is odd, there will be one aggregate containing only a single
+    job and hence users should be careful while defining the parameters for an *aggregate operation*.
+
+Sorting jobs for aggregation
+----------------------------
+
+**signac-flow** allows users to define the sorting order of jobs before creating the aggregates with the
+help of ``sort_by`` parameter and the sorting order can be defined with the help of ``sort_ascending`` parameter.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2, sort_by='temperature', sort_ascending=False)
+    @Project.operation
+    def op5(job1, job2):
+        pass
+
+.. note::
+
+    In the above example, all the jobs will be sorted by the state point parameter ``temperature`` in descending
+    order and then be aggregated as groups of 2.
+
+Selecting jobs for aggregation
+------------------------------
+**signac-flow** allows users to selectively choose which jobs to pass into operation functions.
+
+.. code-block:: python
+
+    @aggregator(select=lambda job: job.sp.temperature > 0)
+    @Project.operation
+    def op6(job1, job2):
+        pass
+
+
+.. _aggregate_id:
+
+Aggregate ID
+============
+
+Similar to the concept of a job id, an aggregate id is a unique hash identifying an aggregate of jobs.
+The aggregate id is sensitive to the order of the jobs in the aggregate
+
+.. note::
+
+    The id of an aggregate containing one job is that job's id.
+
+In order to distinguish between aggregate id and a job id, for an aggregate of more than one job
+the aggregate id of that aggregate will always have a prefix ``agg-``.
+
+Users can generate the aggregate id of an aggregate using :meth:`flow.get_aggregate_id`.
+
+.. tip::
+
+    Users can also pass an aggregate id to the ``--job-id`` command-line flag provided by **signac-flow**
+    in ``run``, ``submit``, and ``exec``.
+
+
+.. _aggregation_with_flow_groups:
+
+Aggregation with FlowGroups
+===========================
+
+In order to associate aggregator object with a :py:class:`FlowGroup`, **signac-flow** provides a
+``aggregator_obj`` parameter in :meth:`~flow.FlowProject.make_group`. By default, no aggregation takes
+place for a :py:class:`FlowGroup`.
+
+.. note::
+
+    Currently, **signac-flow** only allows single :class:`~flow.aggregator` per group, i.e. all the operations present
+    in a :py:class:`FlowGroup` will be using a same :class:`~flow.aggregator` object.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject, aggregator
+
+    class Project(FlowProject):
+        pass
+
+    group = Project.make_group('agg-group', aggregator_obj=aggregator())
```

Comment:
I think the example would be easier to follow if you provided an implementation for such an aggregator class.

Reply:
@csadorf, I am not sure I completely follow what you say. Can you please clarify?